infrastructure/cosmosapp/httpclient.go

```diff
@@ -496,6 +496,78 @@ func (client *HTTPClient) Delegation(
 	return nil, nil
 }
 
+func (client *HTTPClient) AnnualProvisions() (coin.DecCoin, error) {
+	rawRespBody, err := client.request(client.getUrl("mint", "annual_provisions"))
+	if err != nil {
+		return coin.DecCoin{}, err
+	}
+	defer rawRespBody.Close()
+
+	var annualProvisionsResp AnnualProvisionsResp
+	if err := jsoniter.NewDecoder(rawRespBody).Decode(&annualProvisionsResp); err != nil {
+		return coin.DecCoin{}, err
+	}
+
+	annualProvisions, coinErr := coin.NewDecCoinFromString(client.bondingDenom, annualProvisionsResp.AnnualProvisions)
+	if coinErr != nil {
+		return coin.DecCoin{}, fmt.Errorf("error parsing coin from annual provision: %v", annualProvisions)
+	}
+
+	return annualProvisions, nil
+}
+
+func (client *HTTPClient) TotalBondedBalance() (coin.Coin, error) {
+	resp := &ValidatorsResp{
+		MaybePagination: &Pagination{
+			MaybeNextKey: nil,
+			Total:        "",
+		},
+	}
+
+	totalBondedBalance := coin.NewCoin(client.bondingDenom, coin.ZeroInt())
+	for {
+		queryUrl := client.getUrl("staking", "validators")
+		if resp.MaybePagination.MaybeNextKey != nil {
+			queryUrl = fmt.Sprintf(
+				"%s?pagination.key=%s",
+				queryUrl, url.QueryEscape(*resp.MaybePagination.MaybeNextKey),
+			)
+		}
+
+		rawRespBody, statusCode, err := client.rawRequest(queryUrl)
+		if err != nil {
+			return coin.Coin{}, err
+		}
+		defer rawRespBody.Close()
+
+		if decodeErr := jsoniter.NewDecoder(rawRespBody).Decode(&resp); decodeErr != nil {
+			return coin.Coin{}, decodeErr
+		}
+		if resp.MaybeCode != nil && *resp.MaybeCode == 2 {
```

Comment:
Can we put a comment for `.MaybeCode value` `2`?

Reply:
Good point.

Reply:
Done, update it to a constant. Code is better the comment 👍🏻